hcl/hcl.go

```diff
@@ -85,11 +88,48 @@ const (
 	ErrStackInvalidRunOrder     errutil.Error = "invalid stack execution order definition"
 )
 
-func NewConfig(reqversion string) Config {
+// NewConfig creates a new HCL config with absdir as config directory path.
+func NewConfig(absdir string, reqversion string) (Config, error) {
```

Comment:
Is the invariant of the path being absolute guaranteed somewhere else ? Since it is checking if it exists here + if it is a dir I imagine we are missing a check if the path is absolute (type documents the path is absolute, but I could pass relative and it would work I think, but could be missing something stupid as usual).

Reply:
not sure how to properly validate that if both fs and prj paths are absolute. It'd help catch `some/dir` but we rarely use them. I think we should talk about this again, it seems wrong what we're doing like documenting the path must be absolute and doing checks everywhere... But yeah I will add the check here to be consistent with other places. Thanks

Reply:
actually I think it should not be required to be an absolute path here because otherwise, it'll make it impossible to call `hcl.ParseFile("something.hcl")`. I will change the docs.

Reply:
Doesn't [filepath.IsAbs](https://pkg.go.dev/path/filepath#IsAbs) work here ? (regarding "not sure how to properly validate that if both fs and prj paths are absolute")